docs/architecture/application/graphql.md

```diff
@@ -37,3 +38,204 @@ app = OrchestratorCore(base_settings=AppSettings())
 # register SUBSCRIPTION_MODEL_REGISTRY
 app.register_graphql(query=NewQuery)
 ```
+
+## Domain Models Auto Registration for GraphQL
+
+When using the register_graphql function, all products in the SUBSCRIPTION_MODEL_REGISTRY will be automatically converted into GraphQL types.
+The registration process iterates through the list, starting from the deepest product block and working its way back up to the product level.
+
+However, there is a potential issue when dealing with a ProductBlock that references itself, as it can lead to an infinite loop.
+To handle this situation, you must manually create the GraphQL type for that ProductBlock and add it to the GRAPHQL_MODELS list.
```

Comment:
```suggestion
To handle this situation, you must manually create the GraphQL type for that ProductBlock and add it to the `GRAPHQL_MODELS` list.
```

Wrapping code symbols in backticks makes it easier to read, imho

Reply:
forgot it in some places yea